### ISSUE TITLE
How to create NamedPath from Apoc call to use with relationships or nodes function?

### ISSUE THREAD

**Issue by maximelovino** [neo4j-contrib/cypher-dsl#115]
Hello,

I am trying to write a query similar to this one:
```cql
MATCH (node:`Node` {id: 'node_42'})
CALL apoc.path.spanningTree(node, {relationshipFilter: '<rel_filter>', labelFilter: '<label_filter>'}) YIELD path
RETURN DISTINCT node, collect(relationships(path)) as rels, collect(nodes(path)) as nodes
```
The issue for me lies in collecting the relationships and the nodes of the path returned by the apoc spanning tree call. I can't seem to find a way to do it with the DSL as the relationships function requires passing a `NamedPath` in the DSL and I can't  get a `NamedPath` from the apoc call or specify in some way a `NamedPath` of name `path` here. My code currently is the following:

```java
var nodes = Cypher.node("Node").named("node").withProperties("id", Cypher.literalOf("node_42"));
var matchNodes = Cypher.match(nodes);
var apoc = match
        .call("apoc.path.spanningTree")
        .withArgs(
            node.getRequiredSymbolicName(),
            Cypher.mapOf(
                "relationshipFilter",
                Cypher.literalOf(relationshipFilterString),
                "labelFilter",
                Cypher.literalOf(labelFilterString)))
        .yield(Cypher.name("path"));
```
And then from that last apoc variable I want to do my return statement, so I would expect to be able to do something like this:

```java
var returnQuery = apoc.returningDistinct(nodes.getRequiredSymbolicName(), Functions.collect(Functions.relationships(Cypher.name("path"))).as("rels"),Functions.collect(Functions.nodes(Cypher.name("path"))).as("nodes"));
```

But the `Functions.relationships(...)` and `Functions.nodes(...)` expects a `NamedPath` and I can't understand how to get one for my `path` yielded by my apoc call.

Thank you in advance for your help, keep up the good work, I'm really enjoying using this library to build my queries apart from this issue 😄

**Comment by michael-simons** [comment]
Hey @maximelovino thanks your kind feedback!

I added to enhancements here: Supporting named items in the YIELD call and named paths only referencing symbolic name.
This opens up for your use case in the exact variant above and some other ways to phrase this task.

**Comment by maximelovino** [comment]
Wow, thank you very much for the quick fix. Amazing, this is exactly what I was thinking about in terms of implementation. Are you going to make a release with the new feature?

**Comment by michael-simons** [comment]
Release 2020.1.4 is already on central so that it can be included with SDN 6.0.3 on Wednesday. 

Enjoy :)

**Comment by maximelovino** [comment]
Perfect, thank you again for your help 😁
